cmd/sops/main.go

```diff
@@ -695,15 +695,18 @@ func keyGroups(c *cli.Context, file string) ([]sops.KeyGroup, error) {
 	return []sops.KeyGroup{group}, nil
 }
 
+// loadConfig will look for an existing config file, either provided through the command line, or using config.FindConfigFile.
+// Since a config file is not required, a return value of nil, nil is possible
 func loadConfig(c *cli.Context, file string, kmsEncryptionContext map[string]*string) (*config.Config, error) {
 	var err error
 	var configPath string
 	if c.String("config") != "" {
 		configPath = c.String("config")
 	} else {
-		configPath, err = config.FindConfigFile(".")
-		if err != nil {
-			// If we can't find a config file, but we were not request it, assume it does not exist
+		// Ignore config not found errors returned from FindConfigFile since the config file is not mandatory
+		configPath, _ = config.FindConfigFile(".")
+		if configPath == "" {
```

Comment:
Why change from checking the error to checking if configPath is empty? Afaict FindConfigFile only returns an error if no config file is found

Reply:
You're correct. Fixed.
Can you rebase the merge into master? If you don't want the extra commits..

Reply:
Yeah, I can just squash on merge